### DISCUSSION TITLE
is netty server idle time out too long?

### DISCUSSION THREAD

**Discussion by genuinner** [apache/dolphinscheduler#10771] [General]
Netty client sends heartbeat to server to keep channel alive every 6 second, the channel will be closed if failed to send, but Netty server will close the channel in 3min without receiving heartbeat, is it too long?

public static final int NETTY_SERVER_HEART_BEAT_TIME = 1000 * 60 * 3 + 1000;
public static final int NETTY_CLIENT_HEART_BEAT_TIME = 1000 * 6;

**Comment by ruanwenjun** [comment]
> Netty client sends heartbeat to server to keep channel alive every 6 second, the channel will be closed if failed to send, but Netty server will close the channel in 3min without receiving heartbeat, is it too long?
> 
> public static final int NETTY_SERVER_HEART_BEAT_TIME = 1000 * 60 * 3 + 1000; public static final int NETTY_CLIENT_HEART_BEAT_TIME = 1000 * 6;

This should be a bug, we may need to change NETTY_SERVER_HEART_BEAT_TIME to 1000 * 60 * 3 + 1000. cc @CalvinKirs @caishunfeng

**Reply by genuinner** [reply]
Sorry, I am not saying there is a bug, just thought timeout time is too long. But when I disconnect the network between master and worker for 100 second, after the network recover, I found the channel is still alive, so maybe the NETTY_SERVER_HEART_BEAT_TIME is proper, and my question solved, thx.

**Reply by ruanwenjun** [reply]
Are you interested to fix this?

**Reply by genuinner** [reply]
I have no idea, I even don't know why NETTY_SERVER_HEART_BEAT_TIME should be 1000 * 60 * 3 + 1000 ,,ԾㅂԾ,,

**Reply by ruanwenjun** [reply]
> I have no idea, I even don't know why NETTY_SERVER_HEART_BEAT_TIME should be 1000 * 60 * 3 + 1000 ,,ԾㅂԾ,,

The server idle time is 3 * client timeout to handle sometimes the heatbeat loss.

**Reply by genuinner** [reply]
client timeout is 6s, is it more like 30 * client timeout?

**Reply by ruanwenjun** [reply]
Yes, so we need to change the server timeout to 3 * 6s
